playground/serve.go

```diff
@@ -16,27 +16,237 @@
 package playground
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"regexp"
+	"strings"
+	"time"
 
-	"appengine"
-	"appengine/urlfetch"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/datastore"
+	"google.golang.org/appengine/log"
+	"google.golang.org/appengine/memcache"
+	"google.golang.org/appengine/taskqueue"
+	"google.golang.org/appengine/urlfetch"
 )
 
+const (
+	COMPONENTS_URL                 = "https://api.github.com/repos/ampproject/amphtml/git/trees/master?recursive=1"
+	COMPONENTS_MEMCACHE_KEY        = "amp-components-list"
+	COMPONENTS_UPDATE_FREQ_SECONDS = 600
```

Comment:
10 minutes feels very frequent. A daily update should be fine (theoretically weekly in sync with the AMP runtime releases).

Reply:
Done